compiler/optimizations/gpuTransforms.cpp

```diff
@@ -421,11 +427,55 @@ bool GpuizableLoop::determineIfShouldErrorIfNotGpuizable() {
   for_alist(expr, cfl->body) {
     CallExpr *call = toCallExpr(expr);
     if (call && call->isPrimitive(PRIM_ASSERT_ON_GPU)) {
-      return true;
+      return call;
     }
   }
 
-  return false;
+  return nullptr;
+}
+
+static const char* findImmediateStringValue(Expr* expr) {
+  auto symExpr = toSymExpr(expr);
+  if (!symExpr) return nullptr;
+
+  auto tryExtractString = [](SymExpr* stringRef) -> const char* {
+    if (stringRef->symbol()->isImmediate()) {
+      auto immediate = toVarSymbol(stringRef->symbol())->immediate;
+      return immediate->string_value();
+    }
+    return nullptr;
+  };
+
+  if (auto initialString = tryExtractString(symExpr)) {
+    return initialString;
+  }
+
+  // the initial symbol is something like local_string_literal, which is
+  // assigned from an _str_literal. Iterate the SymExprs of this symbol
+  // to find the assignment and thus the original symbol.
+  auto initialSymbol = symExpr->symbol();
+  for_SymbolSymExprs(se, initialSymbol) {
+    if (!se->parentExpr) continue;
+    if (auto ce = toCallExpr(se->parentExpr)) {
+      if (ce->isPrimitive(PRIM_MOVE) && ce->get(1) == se) {
+        auto assignFrom = toSymExpr(ce->get(2));
+        if (assignFrom != nullptr) return tryExtractString(assignFrom);
+      }
+    }
+  }
+
+  return nullptr;
+}
+
+void GpuizableLoop::printNonGpuizableError(CallExpr* assertion, Expr* loc) {
+    debuggerBreakHere();
```

Comment:
Why is this not `gdbShouldBreakHere()`?

Reply:
`debuggerBreakHere` calls `gdbShouldBreakHere`.